src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/AceEditor.java

```diff
@@ -2902,6 +2920,44 @@ private void fireLineWidgetsChanged()
       AceEditor.this.fireEvent(new LineWidgetsChangedEvent());
    }
    
+   private static class ScopeTimer
+   {
+      public ScopeTimer(final AceEditor editor)
+      {
+         timer_ = new Timer()
+         {
+            @Override
+            public void run()
+            {
+               if (row_ >= editor.getRowCount())
+                  return;
+               
+               row_ += 200;
```

Comment:
Is there some change that these incremental updates could cause lagginess? (i.e. does tokenizing 200 rows ever take more than 100ms?)